.github/workflows/performance.yml

```
@@ -0,0 +1,160 @@
+
+      - name: Docker debug information
+        run: |
+          docker -v
+          docker-compose -v
+          docker-compose run --rm mysql mysql --version
+          docker-compose run --rm php php --version
+          docker-compose run --rm php php -m
+          docker-compose run --rm php php -i
+          docker-compose run --rm php locale -a
+
+      - name: Install WordPress
+        run: npm run env:install
+
+      - name: Install WordPress Importer plugin and import mock data
+        run: |
+          npm run env:cli -- plugin install wordpress-importer --activate
+          curl -O https://raw.githubusercontent.com/WPTT/theme-test-data/b9752e0533a5acbb876951a8cbb5bcc69a56474c/themeunittestdata.wordpress.xml
+          npm run env:cli -- import themeunittestdata.wordpress.xml --authors=create
+          rm themeunittestdata.wordpress.xml
+
+      - name: Update permalink structure
+        run: |
+          npm run env:cli -- rewrite structure '/%year%/%monthnum%/%postname%/'
+
+      - name: Install MU plugin
+        run: |
+          mkdir ./build/wp-content/mu-plugins
+          cp ./tests/performance/mu-plugins/server-timing.php ./build/wp-content/mu-plugins/server-timing.php
+
+      - name: Run Performance tests (current commit)
+        run: npm run test:performance
+
+      - name: Print Performance tests results
+        run: "node ./tests/performance/results.js"
+
+      - name: Update the environment to 6.1.1
+        run: |
+          npm run env:cli -- core update --version=6.1.1 --force --path=/var/www/build
+          npm run env:cli -- core version --path=/var/www/build
+
+      - name: Run Performance tests (6.1.1)
+        run: npm run test:performance -- --prefix=base
+
+      - name: Print Performance tests results (6.1.1)
+        run: "node ./tests/performance/results.js --prefix=base"
+
+      - name: Set commit details
+        uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975 # v6.4.0
+        id: commit-timestamp
+        with:
+            github-token: ${{secrets.GITHUB_TOKEN}}
+            script: |
+                const commit_details = await github.rest.git.getCommit({owner: context.repo.owner, repo: context.repo.repo, commit_sha: context.sha});
+                return parseInt((new Date( commit_details.data.author.date ).getTime() / 1000).toFixed(0))
+
+      - name: Publish performance results
+        # Only publish results on pushes to trunk.
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/trunk' }}
+        env:
+            COMMITTED_AT: ${{ steps.commit-timestamp.outputs.result }}
+            CODEHEALTH_PROJECT_TOKEN: ${{ secrets.CODEHEALTH_PROJECT_TOKEN }}
+            HOST_NAME: "codehealth.vercel.app"
+        # The base hash hard coded here is the commit to the WP 6.1.1 tag.
+        run: node ./tests/performance/log-results.js $CODEHEALTH_PROJECT_TOKEN trunk $GITHUB_SHA 031cedc06d612c4a74f7b6d3e77d8145f2954a25 $COMMITTED_AT $HOST_NAME
```

Update these environment values.

```suggestion
            CODEVITALS_PROJECT_TOKEN: ${{ secrets.CODEVITALS_PROJECT_TOKEN }}
            HOST_NAME: "codevitals.run"
        # The base hash hard coded here is the commit to the WP 6.1.1 tag.
        run: node ./tests/performance/log-results.js $CODEVITALS_PROJECT_TOKEN trunk $GITHUB_SHA 031cedc06d612c4a74f7b6d3e77d8145f2954a25 $COMMITTED_AT $HOST_NAME
```

---

I wanted to double check that this update will keep the secret secure. Looking [through the related documentation](https://docs.github.com/en/actions/security-guides/encrypted-secrets#example-using-bash), it seems that we may want to enclose `$CODEVITALS_PROJECT_TOKEN` in double quotes (`"`).

---

I did double check this during out internal testing, and the secrets value is not being output anywhere in the logs that I can see.

---

The purpose of the quotes, I believe, is not to obfuscate the secret but rather to prevent any potential issues with special characters. Which doesn't really affect us currently (personally knowing the value of the secret), but could potentially in the future if it should change.

---

That's helpful. I don't see this token changing often, or at least without our knowledge since it's specifically used for this purpose, so I'm going to resolve this as ok for now.